### ISSUE TITLE
Exported Image IDs are GUIDs using VortoContentMapper

### ISSUE THREAD

**Issue by aDotNetDeveloper** [KevinJump/uSync3.0#115]
Hi Kevin

currently using uSync 3.3.1.740 on Umb 7.5.4

Since enabling _Jumoo.uSync.ContentMappers.VortoContentMapper_, exported image ids have become GUIDs, previously being just the IDs

example of exported property data;

**_previously, without using VortoContentMapper_**
```
 <bannerData><![CDATA[{"values":{"en-US":"[{\"name\":\"Item 1\",\"ncContentTypeAlias\":\"bannerDataItem\",\"bannerImage\":\"1151\",
```
**_using VortoContentMapper_**
```
 <bannerData><![CDATA[{"values":{"en-US":"[\r\n  {\r\n    \"name\": \"Item 1\",\r\n    \"ncContentTypeAlias\": \"bannerDataItem\",\r\n    \"bannerImage\": \"e08d17e1-de81-4fdd-9e89-4ceda4344ebf\",
```
The property on the document type is a _'CTH.ExtendedMediaPicker'_ and the underlying media type is _'Umbraco.ImageCropper'_

The uSync config for _CTH.ExtendedMediaPicker_ is
```
    <uSyncValueMapperSettings>
      <DataTypeId>CTH.ExtendedMediaPicker</DataTypeId>
      <MappingType>media</MappingType>
      <ValueStorageType>text</ValueStorageType>
      <ValueAlias>startNodeId</ValueAlias>
    </uSyncValueMapperSettings>
```

hoping its something wrong in the config, can supply more files and code should you need it.

regards
Wayne

**Comment by KevinJump** [comment]
Hi, just to confirm so are they guids when imported back in ? (because they should be guids in the usync files. that is part of the mapping between installs)

it looks like using the vorto mapper means the serialisation is working (when it wasn't previously) - but it isn't deserialising properly - if the guids aren't going back to Ids

**Comment by aDotNetDeveloper** [comment]
Yes, thinking about this more. being GUIDs is ideal as one of the issues I faced often was the assigned images being incorrect after a fresh import (due to recent doctypes being added etc).

**So, yes they are still GUIDs after import** - is there a generic mechanism that is used to remap ?  I can debug this and see where it needs to be added.  **Edit:** could this failure of remapping be due to the error _Unable to cast object of type 'Newtonsoft.Json.Linq.JArray' to type 'Newtonsoft.Json.Linq.JObject'._ occurring during the import ?

Also, GUIDs for content IDs would be ideal too, as things like references inside Imulus.UrlPicker don't get remapped after import, thus breaking all the links or making them very random.
```
[
  {
    "type": "content",
    "meta": {
      "title": "About",
      "newWindow": false
    },
    "typeData": {
      "url": "",
      "contentId": 1154,
      "mediaId": null
    },
    "disabled": false
  }
]
```
This would need to occur after deserialisation as the referenced content node may not yet exist.

**Comment by KevinJump** [comment]
yeah - if i'm honest the navigation of the grid stuff in json isn't great - it would be nice if there where some objects you could serialize them to in the core - but there isn't.

I will have a look see and see if i can make this bit - a bit more tolerant of random json :)

**Comment by KevinJump** [comment]
There is a quite significant grid mapper update in the pipeline aa1b0b8bd03d376d65db4b771c5d2890bdd50ed7 this coupled with a new media mapper for imagecropper things should resolve this. - but it will need some testing.

**Comment by KevinJump** [comment]
I have tested and I think this is fixed in the latest release 3.3.2

**Comment by aDotNetDeveloper** [comment]
Hi Kevin,

I have found that the image GUIDs will not get mapped to IDs in certain complex nested scenarios.

I'v created a project that uses latest version of Umbraco + uSync and has a simple implementation of a image in a composite, inside nestedcontent inside a vorto

on import of the content page, the image id remains as a guid.- shown in the about page
![image](https://cloud.githubusercontent.com/assets/6894441/23648261/059840f4-036e-11e7-96d6-0540f0865afc.png)

sample project here;
[https://github.com/aDotNetDeveloper/Umbraco-uSync](https://github.com/aDotNetDeveloper/Umbraco-uSync)

regards
Wayne
